src/package-linker.js

```diff
@@ -380,6 +387,10 @@ export default class PackageLinker {
         linkBinConcurrency,
       );
     }
+
+    for (const [, {pkg}] of flatTree) {
+      await this._warnForMissingBundledDependencies(pkg);
```

Comment:
I also don't like the fact that we need to check for file/directory existence later on when we already know they are missing above.